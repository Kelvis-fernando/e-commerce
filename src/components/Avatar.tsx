import useAvatar from "../hooks/useAvatar";
import { AvatarContainer } from "../styles/avatar";

interface AvatarProps {
  name: string;
}

const Avatar = ({ name }: AvatarProps) => {
  const { getInitialLettersFromName } = useAvatar(name);

  return (
    <AvatarContainer>
      <span>{getInitialLettersFromName()}</span>
      <h1>{name}</h1>
    </AvatarContainer>
  );
};

export default Avatar;
