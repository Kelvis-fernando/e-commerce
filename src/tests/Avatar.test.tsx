import React from "react";
import { render } from "@testing-library/react";
import Avatar from "../components/Avatar";

describe("Avatar", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<Avatar name={""} />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
