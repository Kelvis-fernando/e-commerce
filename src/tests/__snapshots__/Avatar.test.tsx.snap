// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe glyMxB"
  >
    <span />
    <h1 />
  </div>
</div>
`;
